pkg/controller/persistentvolume/persistentvolume_provisioner_controller.go

```diff
@@ -0,0 +1,490 @@
+	return c.client.PersistentVolumeClaims(claim.Namespace).Update(claim)
+}
+
+func (c *realControllerClient) UpdatePersistentVolumeClaimStatus(claim *api.PersistentVolumeClaim) (*api.PersistentVolumeClaim, error) {
+	return c.client.PersistentVolumeClaims(claim.Namespace).UpdateStatus(claim)
+}
+
+func (c *realControllerClient) GetKubeClient() client.Interface {
+	return c.client
+}
+
+func keyExists(key string, haystack map[string]string) bool {
+	_, exists := haystack[key]
+	return exists
+}
+
+func isAnnotationMatch(key, needle string, haystack map[string]string) bool {
+	value, exists := haystack[key]
+	if !exists {
+		return false
+	}
+	return value == needle
+}
+
+func isRecyclable(policy api.PersistentVolumeReclaimPolicy) bool {
+	return policy == api.PersistentVolumeReclaimDelete || policy == api.PersistentVolumeReclaimRecycle
+}
+
+// VolumeHost implementation
+// PersistentVolumeRecycler is host to the volume plugins, but does not actually mount any volumes.
+// Because no mounting is performed, most of the VolumeHost methods are not implemented.
+func (c *PersistentVolumeProvisionerController) GetPluginDir(podUID string) string {
+	return ""
+}
+
+func (c *PersistentVolumeProvisionerController) GetPodVolumeDir(podUID types.UID, pluginName, volumeName string) string {
+	return ""
+}
+
+func (c *PersistentVolumeProvisionerController) GetPodPluginDir(podUID types.UID, pluginName string) string {
+	return ""
+}
+
+func (c *PersistentVolumeProvisionerController) GetKubeClient() client.Interface {
+	return c.client.GetKubeClient()
+}
+
+func (c *PersistentVolumeProvisionerController) NewWrapperBuilder(spec *volume.Spec, pod *api.Pod, opts volume.VolumeOptions) (volume.Builder, error) {
+	return nil, fmt.Errorf("NewWrapperBuilder not supported by PVClaimBinder's VolumeHost implementation")
+}
+
+func (c *PersistentVolumeProvisionerController) NewWrapperCleaner(spec *volume.Spec, podUID types.UID) (volume.Cleaner, error) {
+	return nil, fmt.Errorf("NewWrapperCleaner not supported by PVClaimBinder's VolumeHost implementation")
+}
+
+func (c *PersistentVolumeProvisionerController) GetCloudProvider() cloudprovider.Interface {
+	return c.cloud
+}
+
+func (c *PersistentVolumeProvisionerController) GetMounter() mount.Interface {
+	return nil
+}
+
+func (c *PersistentVolumeProvisionerController) GetWriter() io.Writer {
+	return nil
+}
+
+func (c *PersistentVolumeProvisionerController) GetHostName() string {
+	return ""
+}
+
+const (
+	// these pair of constants are used by the provisioner.
+	// The key is a kube namespaced key that denotes a volume requires provisioning.
+	// The value is set only when provisioning is completed.  Any other value will tell the provisioner
+	// that provisioning has not yet occurred.
+	pvProvisioningRequiredAnnotationKey    = "volume.experimental.kubernetes.io/provisioning-required"
+	pvProvisioningCompletedAnnotationValue = "volume.experimental.kubernetes.io/provisioning-completed"
```

Comment:
The value constant is kind of magic, I suppose.  *Any* value other than that constant is considered "not provisioned".

When the value of "requires provisioning" == "provisioning completed", we can move on.

We had talked months ago about using annotations as the interlock between controllers.  I presume this is what you guys had in mind, but I'm open to alternatives.